_std/pathfinding.dm

```diff
@@ -447,7 +447,7 @@
 							continue
 						else
 							return FALSE
-					else //we must be a public door
+					else if (O.allowed(passer)) //even if it has no access, it might not let us pass
```

Comment:
IIRC, in stonepillars/goon-flock#264, extra behavior is added so that floorbots and other robots avoid forcing open doors they should not have access to, namely flock doors.

Reply:
I'll take a look tomorrow, I think that refactor still makes sense even with our changes.

Reply:
Refactor here: https://github.com/stonepillars/goon-flock/pull/315